items:
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  id: _adls_error_exception
  parent: com.microsoft.azure.management.datalake.store.models
  children:
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception.AdlsErrorException()
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception.AdlsErrorException(final String)
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception.AdlsErrorException(final String,final Throwable)
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception.AdlsErrorException(final Throwable)
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception.getBody()
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception.getResponse()
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception.setBody(AdlsError)
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception.setResponse(Response)
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: AdlsErrorException
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsErrorException
  type: Class
  source:
    remote: &o0
      path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
      branch: master
      repo: https://github.com/Azure/azure-sdk-for-java
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 16
  package: com.microsoft.azure.management.datalake.store.models
  summary: <p>Exception thrown for an invalid response with <xref href="com.microsoft.azure.management.datalake.store.models._adls_error" data-throw-if-not-resolved="false"></xref> information. </p>
  syntax: &o1
    content: public class AdlsErrorException
  inheritance:
  - java.lang.Object
  - RestException
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception.AdlsErrorException()
  id: AdlsErrorException()
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: AdlsErrorException()
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsErrorException.AdlsErrorException()
  type: Constructor
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 28
  package: com.microsoft.azure.management.datalake.store.models
  summary: <p>Initializes a new instance of the <xref href="com.microsoft.azure.management.datalake.store.models._adls_error_exception" data-throw-if-not-resolved="false"></xref> class. </p>
  syntax:
    content: public AdlsErrorException()
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception.AdlsErrorException(final String)
  id: AdlsErrorException(final String)
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: AdlsErrorException(final String message)
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsErrorException.AdlsErrorException(final String message)
  type: Constructor
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 34
  package: com.microsoft.azure.management.datalake.store.models
  summary: >-
    <p>Initializes a new instance of the <xref href="com.microsoft.azure.management.datalake.store.models._adls_error_exception" data-throw-if-not-resolved="false"></xref> class.</p>

    <p></p>
  syntax:
    content: public AdlsErrorException(final String message)
    parameters:
    - id: message
      type: 3d93591e
      description: <p>The exception message. </p>
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception.AdlsErrorException(final String,final Throwable)
  id: AdlsErrorException(final String,final Throwable)
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: AdlsErrorException(final String message, final Throwable cause)
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsErrorException.AdlsErrorException(final String message, final Throwable cause)
  type: Constructor
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 43
  package: com.microsoft.azure.management.datalake.store.models
  summary: >-
    <p>Initializes a new instance of the <xref href="com.microsoft.azure.management.datalake.store.models._adls_error_exception" data-throw-if-not-resolved="false"></xref> class.</p>

    <p></p>
  syntax:
    content: public AdlsErrorException(final String message, final Throwable cause)
    parameters:
    - id: message
      type: 3d93591e
      description: <p>the exception message </p>
    - id: cause
      type: fa00b19f
      description: <p>exception that caused this exception to occur </p>
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception.AdlsErrorException(final Throwable)
  id: AdlsErrorException(final Throwable)
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: AdlsErrorException(final Throwable cause)
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsErrorException.AdlsErrorException(final Throwable cause)
  type: Constructor
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 51
  package: com.microsoft.azure.management.datalake.store.models
  summary: >-
    <p>Initializes a new instance of the <xref href="com.microsoft.azure.management.datalake.store.models._adls_error_exception" data-throw-if-not-resolved="false"></xref> class.</p>

    <p></p>
  syntax:
    content: public AdlsErrorException(final Throwable cause)
    parameters:
    - id: cause
      type: fa00b19f
      description: <p>exception that caused this exception to occur </p>
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception.getBody()
  id: getBody()
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: getBody()
  fullName: AdlsError com.microsoft.azure.management.datalake.store.models.AdlsErrorException.getBody()
  type: Method
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 67
  package: com.microsoft.azure.management.datalake.store.models
  summary: >-
    <p>Gets the HTTP response body.</p>

    <p></p>
  syntax:
    content: public AdlsError getBody()
    return:
      type: com.microsoft.azure.management.datalake.store.models._adls_error
      description: <p>the response body </p>
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception.getResponse()
  id: getResponse()
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: getResponse()
  fullName: Response com.microsoft.azure.management.datalake.store.models.AdlsErrorException.getResponse()
  type: Method
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 59
  package: com.microsoft.azure.management.datalake.store.models
  summary: >-
    <p>Gets information about the associated HTTP response.</p>

    <p></p>
  syntax:
    content: public Response getResponse()
    return:
      type: e9d34ed6
      description: <p>the HTTP response </p>
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception.setBody(AdlsError)
  id: setBody(AdlsError)
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: setBody(AdlsError body)
  fullName: void com.microsoft.azure.management.datalake.store.models.AdlsErrorException.setBody(AdlsError body)
  type: Method
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 83
  package: com.microsoft.azure.management.datalake.store.models
  summary: >-
    <p>Sets the HTTP response body.</p>

    <p></p>
  syntax:
    content: public void setBody(AdlsError body)
    parameters:
    - id: body
      type: com.microsoft.azure.management.datalake.store.models._adls_error
      description: <p>the response body </p>
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception.setResponse(Response)
  id: setResponse(Response)
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: setResponse(Response response)
  fullName: void com.microsoft.azure.management.datalake.store.models.AdlsErrorException.setResponse(Response response)
  type: Method
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 75
  package: com.microsoft.azure.management.datalake.store.models
  summary: >-
    <p>Sets the HTTP response.</p>

    <p></p>
  syntax:
    content: public void setResponse(Response response)
    parameters:
    - id: response
      type: e9d34ed6
      description: <p>the HTTP response </p>
references:
- uid: 3d93591e
  spec.java:
  - name: final String
    fullName: final String
- uid: fa00b19f
  spec.java:
  - name: final Throwable
    fullName: final Throwable
- uid: e9d34ed6
  spec.java:
  - name: Response
    fullName: Response
- uid: com.microsoft.azure.management.datalake.store.models._adls_error
  parent: com.microsoft.azure.management.datalake.store.models
  href: com.microsoft.azure.management.datalake.store.models._adls_error.yml
  name: AdlsError
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsError
  type: Class
  summary: <p>Data Lake Store filesystem error containing a specific WebHDFS exception. </p>
  syntax:
    content: public class AdlsError
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  parent: com.microsoft.azure.management.datalake.store.models
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  name: AdlsErrorException
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsErrorException
  type: Class
  summary: <p>Exception thrown for an invalid response with <xref href="com.microsoft.azure.management.datalake.store.models._adls_error" data-throw-if-not-resolved="false"></xref> information. </p>
  syntax: *o1
